src/task_impl/mod.rs

```diff
@@ -105,6 +105,17 @@ pub fn park() -> Task {
     current()
 }
 
+/// Determines if a task is currently being executed.
+///
+/// If this function returns false, `current` and `Task::will_notify_current` will panic.
+pub fn has_current() -> bool {
```

Comment:
Has current doesn't really mean much to me. Maybe `is_task_running`, or `in_task` is a better name.

Reply:
I tend to agree, but my use case is this:

I am using tokio-core to interface an external library that needs to tie into a main loop. Naturally, that library has callbacks into rust code. As long as the main loop runs, these calls are all made from within tasks and there is no problem.

However, once the mainloop shuts down and I call the cleanup methods in the external library, they are not called from within a task. The external library makes callback to rust code during cleanup (unregister events, ...). Since I have no way of knowing that the main loop has stopped, those methods will likely panic.